### ISSUE TITLE
Passed all test cases but "Function" cannot be found

### ISSUE THREAD

**Issue by funkypants5** [nus-cs2030/2324-s2#377]
Hi anybody knows why there this error. I've already imported java.util.function.Function and test cases worked on my computer.

![image](https://github.com/nus-cs2030/2324-s2/assets/70506230/467ca0d5-d286-401c-9742-7702cd3eadbe)

**Comment by santtoshh** [comment]
You have to `import java.util.function.Function;`

**Comment by jpowcm** [comment]
did u `import java.util.function.Function` in this particular file (level5_JS)?

**Comment by santtoshh** [comment]
oh my bad i didnt realise you already tried importing it hahaha. Personally i made use of lambda functions so I didnt need to use the Function interface

**Comment by santtoshh** [comment]
You can also maybe try using lambda functions instead so that you dont have to import the Function interface

**Comment by trishaa17** [comment]
hi , have to import function in level 5.jsh

**Comment by mChewWW** [comment]
import function in all files its used in

**Comment by MoeySeanJean** [comment]
import Function where it is used

**Comment by yushanedi** [comment]
As the others have mentioned, you need to import java.util.function.Function

**Comment by jiancheng37** [comment]
import function wherever u use it

**Comment by joshualek09** [comment]
hi funkypants5, 
if you have already imported Function in your Log file, you dont have to import it in your level 5.
i **think your file name may have been labelled wrongly** as "level5_JS.java" -> and you didnt declare a Function class{} -> thus cannot find the class Function. I have encountered this issue multiple times before, and sometimes its also because in the .java file i accidentally added an extra / forgot to put the closing curly brace at the bottom of the java file

put simply, file name should be of ".jsh" instead of ".java"

**Comment by zzhengweii** [comment]
yup you nreed to have the import line function to use Function

**Comment by cherylkohzw** [comment]
you need to import Function first

**Comment by ZHD1987E** [comment]
I hope all of these worked, @funkypants5.

If you are happy with these, you might want to close this issue and help keep this forum clean yeah?

Thanks!
